src/main/java/com/github/os72/protocjar/maven/ProtocJarMojo.java

```diff
@@ -243,11 +243,52 @@ public void execute() throws MojoExecutionException {
 			if (target.outputDirectorySuffix != null) {
 				target.outputDirectory = new File(target.outputDirectory, target.outputDirectorySuffix);
 			}
+            // Copy files to a temporary directory.
+            target.outputDirectoryFinal = target.outputDirectory;
+            target.outputDirectory = new File(target.outputDirectory.getParent(), target.outputDirectory.getName() + "-tmp");
+
 		}
 		
 		performProtoCompilation();
 	}
 
+    private void copyAndUpdateOutputTarget(OutputTarget outputTarget) throws MojoExecutionException {
+        File src = outputTarget.outputDirectory;
+        File dst = outputTarget.outputDirectoryFinal;
+
+        dst.mkdirs();
+
+        // List files in dst and copy them to target.
+        Collection<File> files = org.apache.commons.io.FileUtils.listFiles(src, null, true);
```

Comment:
Shall i do this in next PR. Since i am committing this to v3.1.0, i want to update protoc to v3.1.0 before i can test new changes.